### PR TITLE
Minor fix rnnt

### PR DESCRIPTION
**Pull request by b-flo**
This is a follow-up to #2727 : 

- Rename `custom_(tt|ct).yaml` to `tdnn_(transformer|conformer)-rnn_transducer.yaml` and update README
- Remove  `warp-rnnt` from `check_install.py`
- Modified `warp-rnnt` install to automatically find cuda version (previously an argument was passed to the script from Makefile)
- Added installation instructions for Warp RNNT to `installation.md`
- Fixed a bug found by @yuekaizhang where an error is raised during inference if `trans-type: warp-rnnt` and no GPU is found. The error is raised when loading trained model, the initialization of transducer loss will fail due to the checking.
I'm not sure if it's the proper way to fix it, I couldn't rely on `self.training` because `self.eval()` is called after loading model.

### CODE REVIEW

File: doc/installation.md [hunk 1]
```diff
@@ -239,9 +239,14 @@ e.g.
     ```sh
     cd <espnet-root>/tools
     . activate_python.sh
-    . ./setup_cuda_env.sh <cuda-root>  # e.g. <cuda-root> = /usr/local/cuda
     ./installers/install_warp-transducer.sh
```
Comment (kamo-naoyuki) [inline doc/installation.md:242]: This line is required

File: tools/check_install.py [hunk 2]
```diff
@@ -101,17 +99,6 @@ def main(args):
             logging.warning("--> %s is not installed.\n###### Raw Error ######\n%s#######################" % (name, traceback.format_exc()))
             is_correct_installed_list.append(False)
 
-    # warp-rnnt was only tested and successfull with CUDA_VERSION=10.0
-    # however the library installation is optional ("warp-transducer" is used by default)
```
Comment (kamo-naoyuki) [inline tools/check_install.py:105]: Why you removed

**Comment by kamo-naoyuki** [comment]
Please split PRs from the next PR.  One PRs can only change one thing.
   (For now, it's okay)

File: doc/installation.md [hunk 3]
```diff
@@ -239,9 +239,14 @@ e.g.
...
     ```
+- To install Warp RNNT
+    ```sh
+    cd <espnet-root>/tools
```
Comment (kamo-naoyuki) [inline doc/installation.md:246]: I think this is not necessary. RNNT is not worth to dare to write

File: doc/installation.md (hunk 1 above)
Comment (b-flo) [reply doc/installation.md:242]: Hum, I can install both `warp-transducer` and `warp-rnnt` without executing `setup_cuda_env.sh`?

File: tools/check_install.py (hunk 2 above)
Comment (b-flo) [reply tools/check_install.py:105]: Because you made `warp-transducer `and` warp-rnnt `optional and `warp-transducer` was already removed from checking. Should I keep it?

File: doc/installation.md (hunk 3 above)
Comment (b-flo) [reply doc/installation.md:246]: OK I'll remove it. However I'll revisit warp-transducer vs warp-rnnt, it seems I have some (small) differences compared to my initial benchmark!

File: doc/installation.md (hunk 1 above)
Comment (kamo-naoyuki) [reply doc/installation.md:242]: It's required.

File: tools/check_install.py (hunk 2 above)
Comment (kamo-naoyuki) [reply tools/check_install.py:105]: Basically, I carefully wrote it. Please keep.

File: doc/installation.md (hunk 3 above)
Comment (kamo-naoyuki) [reply doc/installation.md]: These tools are shown for examples. We don't need to show all.

File: tools/check_install.py (hunk 2 above)
Comment (b-flo) [reply tools/check_install.py:105]: Sorry I'm a bit confused with this one. I further modified to have the whole picture but I'll revert after I understood:

 1) `warp-transducer` (and also `warp-ctc`) checking was already removed before this PR  (lines ares commented)
 1) What I removed are only additions/modifications I introduced in #2444. It concers `warp-transducer` + `warp-rnnt` and some special cases for them: checking in case ESPnet was installed in CPU mode + manual appending to `library_list`(l.106 - l.113) and checking version through dpkg for both rnnt lib because they forgot adding `__version__` in some cases (l.134 - L.141).

Comment (kamo-naoyuki) [reply tools/check_install.py:105]: This is temporary state. Please keep.

Comment (b-flo) [reply tools/check_install.py]: OK, file reverted.

File: doc/installation.md [hunk 4]
```diff
@@ -239,7 +239,7 @@ e.g.
     ```sh
     cd <espnet-root>/tools
     . activate_python.sh
-    . ./setup_cuda_env.sh <cuda-root>  # e.g. <cuda-root> = /usr/local/cuda
+    . ./setup_cuda_env.sh <cuda-root>
```
Comment (kamo-naoyuki) [inline doc/installation.md:242]: Would you revert as it is?

**Review by kamo-naoyuki — APPROVED** [review]
Thanks.
